amaranth/hdl/_mem.py

```diff
@@ -101,7 +96,7 @@ def __getitem__(self, index):
 
         def __setitem__(self, index, value):
             if self._frozen:
-                raise FrozenMemory("Cannot set 'init' on a memory that has already been elaborated")
+                raise AlreadyElaborated("Cannot set 'init' on a memory that has already been elaborated")
```

Comment:
(This diagnostic and the one in the next file over is probably okay as-is, because "modify a memory" is ambiguous on whether it refers to compile-time or run-time modification.)